datafusion/expr/src/expr.rs

```diff
@@ -2263,7 +2267,11 @@ mod test {
             let fun = find_df_window_func(name).unwrap();
             let fun2 = find_df_window_func(name.to_uppercase().as_str()).unwrap();
             assert_eq!(fun, fun2);
-            assert_eq!(fun.to_string(), name.to_uppercase());
+            if fun.to_string() == "first_value" || fun.to_string() == "last_value" {
+                assert_eq!(fun.to_string(), name);
+            } else {
+                assert_eq!(fun.to_string(), name.to_uppercase());
```

Comment:
I wonder, maybe we should treat udf names case insensitive way. What do you think?

Reply:
https://github.com/apache/datafusion/issues/10695 track the issue to rename `name` to lowercase